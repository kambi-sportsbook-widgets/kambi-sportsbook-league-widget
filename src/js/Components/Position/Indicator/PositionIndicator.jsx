import React, { PropTypes } from 'react';
import styles from './PositionIndicator.scss';
import PositionCircle from '../Circle/PositionCircle';

const PositionIndicator = ({ position, color, change }) => {
   return (
      <span>
         <PositionCircle color={color}>{position}</PositionCircle>
         {change !== 0 &&
            <i className={[styles.triangle, change > 0 ? 'up' : 'down'].join(' ')} />}
      </span>
   );
};

PositionIndicator.propTypes = {
   /**
    * Current participant position
    */
   position: PropTypes.number.isRequired,

   /**
    * Participants count
    */
   color: PropTypes.string,

   /**
    * Position change from last statistics publication (positive, negative or 0 number)
    */
   change: PropTypes.number
};

PositionIndicator.defaultProps = {
   change: 0,
   color: null
};

export default PositionIndicator;
