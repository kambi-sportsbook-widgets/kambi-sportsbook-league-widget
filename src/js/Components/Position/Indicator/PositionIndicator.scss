:local(.triangle) {
   margin: 0 10px 0 12px;
   display: inline-block;
   border-left: 5px solid transparent;
   border-right: 5px solid transparent;

   &.up {
      border-bottom: 10px solid #B8E986;
   }

   &.down {
      border-top: 10px solid #FF3A52;
   }
}
