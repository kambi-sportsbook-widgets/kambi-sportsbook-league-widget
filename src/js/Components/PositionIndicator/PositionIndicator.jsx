import React, { PropTypes } from 'react';
import styles from './PositionIndicator.scss';

const positionCircleClassName = function(position, count) {
//   return null;

   if (position >= 1 && position <= 3) {
      return 'green';
   } else if (position === 4) {
      return 'light-green';
   } else if (position === 5) {
      return 'orange';
   } else if (count - position < 3) {
      return 'red';
   } else {
      return null;
   }
};

const PositionIndicator = ({ position, count, change }) => {
   return (
      <span>
         <span className={[styles.position, positionCircleClassName(position, count)].join(' ')}>
            {position}
         </span>
         {change !== 0 &&
            <i className={[styles.triangle, change > 0 ? 'up' : 'down'].join(' ')} />}
      </span>
   );
};

PositionIndicator.propTypes = {
   /**
    * Current participant position
    */
   position: PropTypes.number.isRequired,

   /**
    * Participants count
    */
   count: PropTypes.number.isRequired,

   /**
    * Position change from last statistics publication (positive, negative or 0 number)
    */
   change: PropTypes.number
};

PositionIndicator.defaultProps = {
   change: 0
};

export default PositionIndicator;
