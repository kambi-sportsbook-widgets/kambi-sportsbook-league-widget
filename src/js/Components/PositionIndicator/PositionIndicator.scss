:local(.triangle) {
   margin: 0 10px 0 12px;
   display: inline-block;
   border-left: 5px solid transparent;
   border-right: 5px solid transparent;

   &.up {
      border-bottom: 10px solid #B8E986;
   }

   &.down {
      border-top: 10px solid #FF3A52;
   }
}

:local(.position) {
   border-radius: 50%;
   background: transparent;
   width: 22px;
   height: 22px;
   display: inline-block;
   line-height: 21px;
   text-align: center;
   font-size: 14px;

   &.green {
      color: #fff;
      background: #719D45;
   }

   &.light-green {
      color: #fff;
      background: #8CC354;
   }

   &.orange {
      color: #fff;
      background: #F5A623;
   }

   &.red {
      color: #fff;
      background: #FF3A52;
   }
}
