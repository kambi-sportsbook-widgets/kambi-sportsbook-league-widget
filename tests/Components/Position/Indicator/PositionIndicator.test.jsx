/* eslint-env jest */
import React, { Children } from 'react';
import PositionIndicator from '../../../../src/js/Components/Position/Indicator/PositionIndicator';
import ReactTestRenderer from 'react-test-renderer';

describe('PositionIndicator DOM rendering', () => {

   it('renders correctly with default props', () => {
      const tree = ReactTestRenderer.create(
         <PositionIndicator position={1} />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with color', () => {
      const tree = ReactTestRenderer.create(
         <PositionIndicator position={1} color="GREEN" />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with positive position change', () => {
      const tree = ReactTestRenderer.create(
         <PositionIndicator position={1} change={5} />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly with negative position change', () => {
      const tree = ReactTestRenderer.create(
         <PositionIndicator position={4} change={-2} />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

});
