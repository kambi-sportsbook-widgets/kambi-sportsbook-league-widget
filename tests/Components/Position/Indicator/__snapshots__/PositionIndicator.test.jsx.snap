// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionIndicator DOM rendering renders correctly with color 1`] = `
<span>
  <span
    className="wrapper false"
  >
    <span
      className="circle KambiWidget-box--win"
    >
      1
    </span>
  </span>
</span>
`;

exports[`PositionIndicator DOM rendering renders correctly with default props 1`] = `
<span>
  <span
    className="wrapper no-circle"
  >
    <span
      className="circle transparent"
    >
      1
    </span>
  </span>
</span>
`;

exports[`PositionIndicator DOM rendering renders correctly with negative position change 1`] = `
<span>
  <span
    className="wrapper no-circle"
  >
    <span
      className="circle transparent"
    >
      4
    </span>
  </span>
  <i
    className="triangle down"
  />
</span>
`;

exports[`PositionIndicator DOM rendering renders correctly with positive position change 1`] = `
<span>
  <span
    className="wrapper no-circle"
  >
    <span
      className="circle transparent"
    >
      1
    </span>
  </span>
  <i
    className="triangle up"
  />
</span>
`;
